### ISSUE TITLE
Error updating Astronvim

### ISSUE THREAD

**Issue by tariqbaater** [AstroNvim/AstroNvim#1872]
### Checklist

- [X] Using a stable version of Neovim (i.e. not neovim nightly)
- [X] `:AstroUpdate`
- [X] Restarted AstroNvim

### Operating system/version

macOs 13.3.1

### Terminal/GUI

Terminal

## Astronvim
- AstroNvim Version: v3.10.3
-  Neovim Version: v0.9.0
- OK Using stable Neovim >= 0.8.0

### Describe the bug

Error running command: git -C /Users/username/.config/nvim config branch.main.remote
Error message:

Error executing Lua callback: ...s/username/.config/nvim/lua/astronvim/utils/git.lua:76: attempt to concatenate local 'remote' (a nil value)
stack traceback:
        ...s/username/.config/nvim/lua/astronvim/utils/git.lua:76: in function 'remote_url'
        ...username/.config/nvim/lua/astronvim/utils/updater.lua:123: in function 'create_rollback'
        ...username/.config/nvim/lua/astronvim/utils/updater.lua:256: in function 'update'
        ...rs/username/.config/nvim/lua/astronvim/autocmds.lua:237: in function <...rs/username/.config/nvim/lua/astronvim/autocmds.lua:237>

### Steps to Reproduce

I typed :AstroUpdate command in the nvim command line and it produced the above error.

### Expected behavior

I was expecting a normal update of new updated files being pulled from the AstroNvim repo.

### Screenshots

<img width="1369" alt="Screenshot 1444-10-18 at 11 58 39 AM" src="https://user-images.githubusercontent.com/23321048/236782542-4be4be73-7467-408a-b959-4eaa91b40bd0.png">

**Comment by mehalter** [comment]
What version are you currently on?

**Comment by mehalter** [comment]
Can you share your user configuration?

**Comment by tariqbaater** [comment]
return {
  -- Configure AstroNvim updates
  updater = {
    remote = "origin", -- remote to use
    channel = "stable", -- "stable" or "nightly"
    version = "latest", -- "latest", tag name, or regex search like "v1.*" to only do updates before v2 (STABLE ONLY)
    branch = "nightly", -- branch name (NIGHTLY ONLY)
    commit = nil, -- commit hash (NIGHTLY ONLY)
    pin_plugins = nil, -- nil, true, false (nil will pin plugins on stable only)
    skip_prompts = false, -- skip prompts about breaking changes
    show_changelog = true, -- show the changelog after performing an update
    auto_quit = false, -- automatically quit the current session after a successful update
    remotes = { -- easily add new remotes to track
    },
  },

**Comment by tariqbaater** [comment]
> Can you share your user configuration?

return {
  -- Configure AstroNvim updates
  updater = {
    remote = "origin", -- remote to use
    channel = "stable", -- "stable" or "nightly"
    version = "latest", -- "latest", tag name, or regex search like "v1.*" to only do updates before v2 (STABLE ONLY)
    branch = "nightly", -- branch name (NIGHTLY ONLY)
    commit = nil, -- commit hash (NIGHTLY ONLY)
    pin_plugins = nil, -- nil, true, false (nil will pin plugins on stable only)
    skip_prompts = false, -- skip prompts about breaking changes
    show_changelog = true, -- show the changelog after performing an update
    auto_quit = false, -- automatically quit the current session after a successful update
    remotes = { -- easily add new remotes to track
    },
  },

**Comment by mehalter** [comment]
What do you get when you run the following commands in your `~/.config/nvim` folder?

```
git rev-parse --abbrev-ref HEAD
git config branch.main.remote
git remove get-url origin
```

**Comment by tariqbaater** [comment]
I think the last command was supposed to be git remote, but i get the below repo url.

"https://github.com/AstroNvim/AstroNvim"

**Comment by mehalter** [comment]
you are correct, what are the results you get from the other 2 commands as well?

**Comment by tariqbaater** [comment]
> you are correct, what are the results you get from the other 2 commands as well?

the first one returned "HEAD" and the second one nothing happened.

**Comment by mehalter** [comment]
when you run the 2nd one and it returns nothing, what do you get when you run `echo $?` basically what is the exit code?

**Comment by mehalter** [comment]
what do you get when you run `git ---version`?

**Comment by tariqbaater** [comment]
> what do you get when you run `git ---version`?

i get nothing when i echo the last command, when i run  git --version i get "git version  2.39.2 (Apple Git-143)".
my git works well i have pushed several commits and even the changes that i have updated on my other init files.

**Comment by tariqbaater** [comment]
Should i just uninstall and re-install because i can pull back all my initial changes and configurations, i was thinking maybe there is another way i can resolve the issue.

**Comment by mehalter** [comment]
this looks like there is a problem with that command for some reason returning `nil`. Could you try changing line `122` in `~/.config/nvim/lua/astronvim/utils/updater.lua` to

```lua
  snapshot.remote = git.branch_remote(snapshot.branch, false) or "origin"
```

and then run the update?

**Comment by tariqbaater** [comment]
> ```lua
> snapshot.remote = git.branch_remote(snapshot.branch, false) or "origin"
> ```

This worked, thank you for your quick response really i appreciate it.

**Comment by mehalter** [comment]
Ok I just released this change with v3.15.3. You will probably need to make this change again on that line to get the newest update but >=v3.15.3 that you shouldn't need to

**Comment by tariqbaater** [comment]
> Ok I just released this change with v3.15.3. You will probably need to make this change again on that line to get the newest update but >=v3.15.3 that you shouldn't need to

I will note that down, thanks again mate.
